blog/2023-04-18-magazine-003.md

```diff
@@ -0,0 +1,325 @@
+
+So far we've added support for this resource on both the `tf-aws` and `sim` targets.
+When you compile to AWS, this will produce an AWS API Gateway that exposes the endpoints on the public internet -- and when you compile to the simulator, you'll get a local HTTP server that you can interact with using the Wing Console, or any other HTTP client like `curl` or Postman.
+
+Check out our [API Reference](https://docs.winglang.io/reference/sdk#api-) for more details (docs with examples coming soon), and give us a +1 on any of these issues if you would like support for `cloud.Api` on [GCP](https://github.com/winglang/wing/issues/624), [Azure](https://github.com/winglang/wing/issues/625), or if you'd like to see a dedicated [`cloud.GraphQL` resource](https://github.com/winglang/wing/issues/2155)!
+
+#### `cloud.Bucket` events
+
+Acting on events is a great way to build reactive applications that leverage the cloud's scalability.
+
+Wing's `cloud.Bucket` resource now supports the following events that allow you to run an inflight function when a certain event occurs:
+
+* `on_create` - triggered when a new object is added to the bucket
+* `on_delete` - triggered when an object is deleted from the bucket
+* `on_update` - triggered when an object is updated in the bucket
+* `on_event` - triggered when any of the above events occur
+
+Here's an example of how you can use these events to count the number of objects added to a bucket, and expose that count through an API endpoint:
+
+```js
+bring cloud;
+
+let bucket = new cloud.Bucket();
+let counter = new cloud.Counter();
+
+bucket.on_create(inflight (key: str) => {
+  log(key);
+  counter.inc();
+});
+
+let api = new cloud.Api();
+api.get("/count", inflight (req: cloud.ApiRequest): cloud.ApiResponse => {
+  return cloud.ApiResponse {
+    body: Json { objects_added: counter.peek() },
+    status: 200,
+  };
+});
+```
+
+#### Potpourri of new resources and methods
+
+We've got too many new resources and additions to list them all here, but here are some of the highlights:
+
+- `cloud.Table` - a NoSQL table that can be used to statefully manage structured information supports basic CRUD operations. Only supported on the `tf-aws` target right now. (Note: as databases are hard to abstract, this resource is highly experimental right now and is likely to change in the future.)
+- `cloud.Secret` - a key-value store that can be used to store sensitive information like passwords and API keys. Currently supported on the `sim` and `tf-aws` targets.
+- `redis.Redis` - a Redis instance that can be used to store unstructured data. Currently supports on the `sim` and `tf-aws` targets. When you compile to AWS, this will create a Redis instance in Elasticache hosted in its own VPC, with a security group that only allows access from the VPC that your Wing app is running in. When you compile to the simulator, this resource will use Docker to run a local Redis instance.
```

Comment:
```suggestion
- `redis.Redis` - a Redis instance that can be used to store unstructured data. Currently supported on the `sim` and `tf-aws` targets. When you compile to AWS, this will create a Redis instance in ElastiCache hosted in its own VPC, with a security group that only allows access from the VPC that your Wing app is running in. When you compile to the simulator, this resource will use Docker to run a local Redis instance.
```